pkg/util/deployment/deployment.go

```diff
@@ -123,6 +123,14 @@ func GetNewRCTemplate(deployment extensions.Deployment) api.PodTemplateSpec {
 	return newRCTemplate
 }
 
+// GetRCsForDeployment returns all the replication controllers for the provided deployment.
+func GetRCsForDeployment(c client.Interface, deployment extensions.Deployment) ([]api.ReplicationController, error) {
```

Comment:
I dont think that this function does what you expect it to do.
You want the RCs that have the same selector as deployment.Spec.Selector, not the ones selected by it.

For ex: it will fail for the following RCs and deployment (it will not return the RC it should).

Deployment {
 "spec" : {
     "selector": {
         "pod": "true",
      }
  }
}

RC {
  "labels": {
       "rc": "true",
   }
   "selector": {
     "pod": "true",
  }
}

You can use GetOldRCs and GetNewRCs instead.